apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender.cs

```diff
@@ -194,6 +201,38 @@ public override void ActivateOptions()
             _isActivated = true;
         }
 
+        private LoggingServiceV2Client BuildLoggingServiceClient()
+        {
+            GoogleCredential credential;
+            if(!TryGetCredentialFromFile(out credential))
+            {
+                return LoggingServiceV2Client.Create();
+            }
+            
+            return LoggingServiceV2Client.Create(new Grpc.Core.Channel(LoggingServiceV2Client.DefaultEndpoint.Host, LoggingServiceV2Client.DefaultEndpoint.Port, Grpc.Auth.GoogleGrpcCredentials.ToChannelCredentials(credential)));
+        }
+
+        private bool TryGetCredentialFromFile(out GoogleCredential credential)
+        {
+            credential = null;
+            if(string.IsNullOrWhiteSpace(CredentialFile))
+            {
+                return false;
+            }
+            
+            using (FileStream credStream = File.OpenRead(CredentialFile))
+            {
+                credential = GoogleCredential.FromStream(credStream);

```

Comment:
Please use the new `GoogleCredential.FromFile()` method; available in Google.Apis.Auth from v1.29

Reply:
It may be worth holding off for a week until we've rolled out the use of 1.29.1 - at the moment this still depends on 1.28.

Reply:
I've made the other changes. I'll watch for the 1.29.1 roll out so I can switch to the FromFile method and push an update.

Reply:
Great, thanks. Apologies for not realising that this code hasn't yet upgraded to v1.29!

Reply:
An update on this. I noticed that 1.29.1 is out, however it looks like I have to wait until the Google.Api.Gax.Grpc package is updated to use it since I believe that's where the Google.Api.Auth reference is coming from and even the 2.1.0-beta02 version of that package still uses 1.28. It's not impossible that this could be worked around by applying the Google.Api.Auth package directly but I'd prefer not to get into NuGet package fiddling if I can avoid it. It's far to easy to inadvertently break something.

Reply:
The dependencies are now updated - if you rebase to master you should be able to use it without any change in dependencies.

Reply:
If I open something like the Google.Cloud.Storage.V1 I can use it from there but Google.Cloud.Logging.Log4Net is still using the old version of Google.Apis.Auth.  I think that comes down to Google.Cloud.Logging.V2 still referencing Google.Api.Gax.Grpc 2.0.0 instead of the new Google.Api.Gax.Grpc 2.1.0.

Reply:
Okay, all the changes should be in now. Let me know if you spot anything else that should change.